format_check.py

```diff
@@ -10,35 +10,38 @@
 	for file in files:
 		#print os.path.join(subdir, file)
 		filepath = subdir + os.sep + file
-		if filepath.startswith("./compiler"): continue
-		if filepath.startswith("./saves"): continue
-		if filepath.startswith("./serverAssets"): continue
-		if filepath.startswith("./zig-cache"): continue
-		if filepath.startswith("./.zig-cache"): continue
+		if filepath.startswith(f".{os.sep}compiler"): continue
+		if filepath.startswith(f".{os.sep}saves"): continue
+		if filepath.startswith(f".{os.sep}serverAssets"): continue
+		if filepath.startswith(f".{os.sep}zig-cache"): continue
+		if filepath.startswith(f".{os.sep}.zig-cache"): continue
 
 		if filepath.endswith(".json") or filepath.endswith(".zig") or filepath.endswith(".py") or filepath.endswith(".zon") or filepath.endswith(".vs") or filepath.endswith(".fs") or filepath.endswith(".glsl"):
-			with open(filepath, "r", newline = '') as f:
-				string = f.read()
-				line = 1
-				lineStart = True
-				for i, c in enumerate(string):
-					if(c == '\r'):
-						print("Incorrect line ending \\r in file ", filepath, " in line ", line, ". Please configure your editor to use LF instead of CRLF.")
-						fails += 1
-					elif(c == '\n'):
-						line += 1
-						lineStart = True
-					elif(c == ' '):
-						if(lineStart):
-							print("Incorrect indentation in file ", filepath, " in line ", line, ". Please use tabs instead of spaces.")
+			try:
+				with open(filepath, "r", newline = '', encoding="utf-8") as f:
+					string = f.read()
+					line = 1
+					lineStart = True
+					for i, c in enumerate(string):
+						if(c == '\r'):
+							print("Incorrect line ending \\r in file ", filepath, " in line ", line, ". Please configure your editor to use LF instead of CRLF.")
 							fails += 1
-							lineStart = False # avoid repeating this error multiple times
-					elif(c == '\t'):
-						continue
-					elif(lineStart):
-						lineStart = False
-					else:
-						continue
+						elif(c == '\n'):
+							line += 1
+							lineStart = True
+						elif(c == ' '):
+							if(lineStart):
+								print("Incorrect indentation in file ", filepath, " in line ", line, ". Please use tabs instead of spaces.")
+								fails += 1
+								lineStart = False # avoid repeating this error multiple times
+						elif(c == '\t'):
+							continue
+						elif(lineStart):
+							lineStart = False
+						else:
+							continue
+			except:
+				print("Exception when reading file ", filepath, ": ", traceback.format_exc())
```

Comment:
Did not notice the else: continue, that was there in the original script too. 

Had an exception happen when the program tried to read files like Boulder.zig due to the non-ascii symbols in it. Without the try catch I didn't know if the exception happened due to a file I had modified or if it was an issue with format_check. 

Since I added the explicit encoding to open the try catch should in theory not be needed anymore, so it can be removed again.